eve/main.yml

```diff
@@ -277,7 +284,34 @@ stages:
       - Upload:
           name: upload artifacts
           source: sosreport
+      - ShellCommand:
+          name: Retrieve ISO version N-1 image
+          command: >
+            isourl="/pre/metalk8s-%(prop:metalk8s_prev_version)s.iso";
+            curl -s -XGET -o /tmp/metalk8s.iso
+            "%(prop:artifacts_private_url)s$isourl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Retrieve ISO N-1 image checksum
+          command: >
+            shasumurl="/pre/SHA256SUM-%(prop:metalk8s_prev_version)s";
+            curl -s -XGET -o /tmp/SHA256SUM
+            "%(prop:artifacts_private_url)s$shasumurl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check N-1 image with checksum
+          command: cd /tmp && sha256sum -c SHA256SUM
+          haltOnFailure: true
```

Comment:
IMO you should retrieve and check both ISO at the very beginning (and also you should use the same yaml tag for upgrade and downgrade)